### DISCUSSION TITLE
Problems decoding iBoost+ signals

### DISCUSSION THREAD

**Discussion by rowanbradley** [merbanan/rtl_433#2932] [Q&A]
I am trying to use rtl_433 with protocol 188 to decode the transmissions of my Marlec iBoost+. I have attached a serction of log. Questions:
 - is this protocol meant to decode transmissions from the iBoost to the Buddy, or only the transmissions from the CT sender to the iBoost? It is the former that I need.
 - If this protocol is only intended to read the transmissions from the sender, what would be involved in adding the transmissions to the Buddy?
 - Why do I get several copies of identical readings at the identical time?
 - The values in each record seem to be only mic, mod, freq1, freq2 , rssi, snr, noise, i.e. values about the level and quality of the radio signal received. There is no real payload. Why not? How do I get this?
[iBoost.log](https://github.com/merbanan/rtl_433/files/15393327/iBoost.log)

Thank you - Rowan

**Comment by zuckschwerdt** [comment]
All data uis received, decoded and checked, then output as "raw" hex. Only the frame type 0x22 has been analyzed, your frames seem to be type 0xcd. See https://github.com/merbanan/rtl_433/blob/master/src/devices/marlec_solar.c for all the knowledge we have so far.

Perhaps you can table the raw message data you get with the values you expect to see, then we can try to pick the data apart.

Repeated messages are always shown by rtl_433, there is not duplicate filtering.

**Comment by rowanbradley** [comment]
Thank you very much for your helpful reply. I will try to do what you say. So far I'm not having much luck. Many frames do not seem to contain one of the expected frame-type bytes. And can you examine how gloating point values are encoded? For example, what is 561.60 in hex?

Thank you - Rowan

⁣Get BlueMail for Android ​

**Comment by zuckschwerdt** [comment]
We usually see fixed point values, e.g. 561.6 might be 5616 = 0x15F0. But divisor and offset might be unexpected. Best approach is to ramp a value up and down to spot where the changing bits are, then work out a formula.
